File: cpp_trainings/src/range_sensor_pub.cpp
#include "rclcpp/rclcpp.hpp"
#include "tutorial_interfaces/msg/range_sensor.hpp"
 
class RangeSensorPub : public rclcpp::Node 
{
public:
    RangeSensorPub() : Node("range_sensor_pub") 
    {
        RCLCPP_INFO(this->get_logger(),"Range Sensor publisher node has been started..");
        publisher_ = this->create_publisher<tutorial_interfaces::msg::RangeSensor>("range_sensor",10);
        timer_ = this->create_wall_timer(std::chrono::milliseconds(500),
            std::bind(&RangeSensorPub::callbackSensor,this));
    }
 
private:
    void callbackSensor()
    {   
        auto range_value = tutorial_interfaces::msg::RangeSensor();
        int value = std::rand() % 100 + 1;
        range_value.range = value;
        RCLCPP_INFO(this->get_logger(), "A range value is publishing .. ");
        publisher_->publish(range_value);

    }
    rclcpp::Publisher<tutorial_interfaces::msg::RangeSensor>::SharedPtr publisher_;
    rclcpp::TimerBase::SharedPtr timer_;
};
 
int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<RangeSensorPub>(); 
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: python_trainings/python_trainings/range_sensor_pub.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from tutorial_interfaces.msg import RangeSensor
import random
 
 
class RangeSensorPub(Node): 
    def __init__(self):
        super().__init__("Range_Sensor_Pub")
        
        self.publisher_ = self.create_publisher(RangeSensor, "RangeSensor",10)
        self.timer_ = self.create_timer(0.5,self.callback_range_sensor)

    def callback_range_sensor(self):
        msg = RangeSensor()
        msg.range = random.randint(0,100)
        self.publisher_.publish(msg)
 
 
def main(args=None):
    rclpy.init(args=args)
    node = RangeSensorPub()
    rclpy.spin(node)
    rclpy.shutdown()
 
 
if __name__ == "__main__":
    main()